sdk/python/feast/pyspark/historical_feature_retrieval_job.py

```diff
@@ -1,19 +1,318 @@
+import argparse
 import json
-from datetime import timedelta
-from typing import Any, Dict, List
+from datetime import datetime, timedelta
+from typing import Dict, List, NamedTuple, Optional
 
-from pyspark import SparkFiles
 from pyspark.sql import DataFrame, SparkSession, Window
 from pyspark.sql.functions import col, expr, monotonically_increasing_id, row_number
 
 
+class EntitySource(NamedTuple):
+    """
+    Source for the entity dataframe. The values for the `format` and `path` should be
+    recognizable by the Spark cluster where the job is going to run on. For example, s3a
+    connector is required for Amazon S3 path. Spark avro package should be on the class
+    path of the spark cluster if the format is `avro`.
+
+    Attributes:
+        format (str): File format.
+        path (str): uri to the file source.
+        timestamp_column (str): Column representing the event timestamp.
+        mapping (Dict[str, str]): Optional. If present, the source column will be renamed
+            based on the mapping.
+        options (Dict[str, str]): Spark read options.
+    """
+
+    format: str
+    path: str
+    timestamp_column: str
+    mapping: Dict[str, str] = {}
+    options: Dict[str, str] = {}
+
+
+class FeatureTableSource(NamedTuple):
```

Comment:
I don't understand why we have two sources, nor why `created_timestamp` is needed for the `FeatureTableSource`